tests/study/storage/variantstudy/test_variant_study_service.py

```diff
@@ -239,3 +242,181 @@ def test_generate_task(
+        admin_user = User(id=ADMIN_ID, name=ADMIN_NAME)
+        db.session.add(admin_user)
+        db.session.commit()
+
+        regular_user = User(id=99, name="regular")
+        db.session.add(regular_user)
+        db.session.commit()
+
+        # noinspection PyArgumentList
+        group = Group(id="my-group", name="group")
+        db.session.add(group)
+        db.session.commit()
+
+        # Create a raw study (root of the variant)
+        raw_study_path = tmp_path / "My RAW Study"
+        # noinspection PyArgumentList
+        raw_study = RawStudy(
+            id="my_raw_study",
+            name=raw_study_path.name,
+            version="860",
+            author="John Smith",
+            created_at=datetime.datetime(2023, 7, 15, 16, 45),
+            updated_at=datetime.datetime(2023, 7, 19, 8, 15),
+            last_access=datetime.datetime.utcnow(),
+            public_mode=PublicMode.FULL,
+            owner=admin_user,
+            groups=[group],
+            path=str(raw_study_path),
+            additional_data=StudyAdditionalData(author="John Smith"),
+        )
+
+        db.session.add(raw_study)
+        db.session.commit()
+
+        # Set up the Raw Study
+        raw_study_service.create(raw_study)
+
+        # Variant studies
+        variant_list = []
+
+        # For each variant created
+        with monkeypatch.context() as m:
+            # Set the system date older to create older variants
+            m.setattr("antarest.study.storage.variantstudy.variant_study_service.datetime", FakeDatetime)
+            m.setattr("antarest.study.service.datetime", FakeDatetime)
+
+            for index in range(3):
+                variant_list.append(
+                    variant_study_service.create_variant_study(
+                        raw_study.id,
+                        "Variant{}".format(str(index)),
+                        params=Mock(
+                            spec=RequestParameters,
+                            user=DEFAULT_ADMIN_USER,
+                        ),
+                    )
+                )
+
+                # Generate a snapshot for each variant
+                variant_study_service.generate(
+                    sanitize_uuid(variant_list[index].id),
+                    False,
+                    False,
+                    params=Mock(
+                        spec=RequestParameters,
+                        user=Mock(spec=JWTUser, id=regular_user.id, impersonator=regular_user.id),
+                    ),
+                )
+
+                variant_study_service.get(variant_list[index])
+
+        variant_study_path = Path(tmp_path).joinpath("internal_studies")
+
+        # Check if everything was correctly initialized
+        assert len(list(variant_study_path.iterdir())) == 3
+
+        for variant in variant_study_path.iterdir():
+            assert variant.is_dir()
+            assert list(variant.iterdir())[0].name == "snapshot"
+
+        # Begin tests
```

Comment:
Perhaps just separate the set up and the tests with a bigger sign to ensure better readability. In other tests, we used something like this:
```python
# =============================
#  SET UP
# ============================= 
```